server/historian/packages/historian-base/src/services/redisCache.ts

```diff
@@ -5,13 +5,26 @@
 
 import { Redis } from "ioredis";
 import * as winston from "winston";
-import { ICache } from "./definitions";
+import { ICache, IRedisParameters } from "./definitions";
 
 /**
  * Redis based cache client
  */
 export class RedisCache implements ICache {
-    constructor(private readonly client: Redis, private readonly prefix = "git") {
+    private readonly expireAfterSeconds: number = 60 * 60 * 24;
+    private readonly prefix: string = "git";
+
+    constructor(
+        private readonly client: Redis,
+        parameters?: IRedisParameters) {
+        if (parameters?.expireAfterSeconds) {
+            this.expireAfterSeconds = parameters.expireAfterSeconds;
+        }
+
+        if (parameters?.prefix) {
```

Comment:
Do we use this new prefix anywhere? Or will this be a part of a future change?

Reply:
Good question. Before I made these changes, the prefix was a constructor parameter but we never override it. I would guess we don't plan to make it overridable, since each Redis client wrapper deals with a different type of information. One advantage of not exposing this prefix would be that I can just drop the `IRedisParameters` interface - I wouldn't need it since the only parameter would be the expiry/TTL.

@tanviraumi, how do you feel about not allowing the prefix to be overridden? Do we have any scenarios where we might want to share a Redis client wrapper between 2 "prefixes"? (for example, the RedisCache used to cache summary/Git data would be used for something other than Git?) It doesn't feel like that's the case since we have different wrappers for different types of data (RedisThrottlerManager, RedisTenantManager, RedisClientCache...)

Reply:
I think for future sake, its better to keep it overridable